### PR TITLE
ALL-3936/Package restructure

### PR DESCRIPTION
**Pull request by Mecharyry**
## Problem
Lots and lots of public classes that could be package if they were in the right place.

## Solution
Move classes to logically grouped areas and make them package scope.

### Test(s) added 
No, just updating packages.

### Screenshots
No UI changes.

### Paired with 
@ouchadam

### CODE REVIEW

File: core/src/main/java/com/novoda/noplayer/exoplayer/playererror/ExoPlayerErrorFactory.java [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
-import com.novoda.noplayer.exoplayer.playererror.DrmDecryptionError;
-import com.novoda.noplayer.exoplayer.playererror.DrmRequestError;
-import com.novoda.noplayer.exoplayer.playererror.InvalidResponseCodeError;
-import com.novoda.noplayer.exoplayer.playererror.MalformedContentError;
-import com.novoda.noplayer.exoplayer.playererror.UnknownError;
 
 import java.io.IOException;
 
-final class ExoPlayerErrorFactory {
+public final class ExoPlayerErrorFactory {
```
Comment (zegnus) [inline core/src/main/java/com/novoda/noplayer/exoplayer/playererror/ExoPlayerErrorFactory.java]: Is this ok that is available from the client?

File: core/src/main/java/com/novoda/noplayer/model/TextCues.java [hunk 2]
```diff
@@ -12,7 +12,7 @@ public TextCues(List<Cue> cues) {
         this.cues = cues;
     }
 
-    List<Cue> cues() {
+    public List<Cue> cues() {
```
Comment (zegnus) [inline core/src/main/java/com/novoda/noplayer/model/TextCues.java]: This class needs to be public facing?

File: core/src/main/java/com/novoda/noplayer/exoplayer/playererror/ExoPlayerErrorFactory.java (hunk 1 above)
Comment (Mecharyry) [inline core/src/main/java/com/novoda/noplayer/exoplayer/playererror/ExoPlayerErrorFactory.java]: I don't see the problem with this being accessible from the client. They will only get a specific `Player.PlayerError` based on an `Exception`. They can't extend it or anything else.

File: core/src/main/java/com/novoda/noplayer/model/TextCues.java (hunk 2 above)
Comment (Mecharyry) [inline core/src/main/java/com/novoda/noplayer/model/TextCues.java]: It wouldn't need to be if we collapsed some more packages but all of our models now belong in the `model` package meaning they need to be `public`. Personally, I would find it strange if we didn't expose our `NoPlayer` models.

File: core/src/main/java/com/novoda/noplayer/exoplayer/playererror/ExoPlayerErrorFactory.java (hunk 1 above)
Comment (zegnus) [inline core/src/main/java/com/novoda/noplayer/exoplayer/playererror/ExoPlayerErrorFactory.java]: If this is public a client can do `ExoPlayerErrorFactory.errorFor(...)`. Is this part of the public API?

File: core/src/main/java/com/novoda/noplayer/model/TextCues.java (hunk 2 above)
Comment (zegnus) [inline core/src/main/java/com/novoda/noplayer/model/TextCues.java]: If the client needs to have those models then ok, but if it doesn't then these models become a contract with the clients

File: core/src/main/java/com/novoda/noplayer/exoplayer/playererror/ExoPlayerErrorFactory.java (hunk 1 above)
Comment (Mecharyry) [inline core/src/main/java/com/novoda/noplayer/exoplayer/playererror/ExoPlayerErrorFactory.java]: We opted to rename as `Mapper` and move to the package where it is used 6b400cf

File: core/src/main/java/com/novoda/noplayer/model/TextCues.java (hunk 2 above)
Comment (ouchadam) [inline core/src/main/java/com/novoda/noplayer/model/TextCues.java]: `TextCues` is part of the `NoPlayer` contract as it's used in `PlayerView`

Comment (zegnus) [inline core/src/main/java/com/novoda/noplayer/model/TextCues.java]: ok

File: core/src/test/java/com/novoda/noplayer/exoplayer/ExoPlayerFacadeTest.java [hunk 3]
```diff
@@ -391,7 +391,7 @@ public void whenGettingAudioTracks_thenDelegatesToTrackSelector() {
         @Mock
         DrmSessionCreator drmSessionCreator;
         @Mock
-        ExoPlayerDrmSessionEventListener drmSessionEventListener;
+        DefaultDrmSessionManager.EventListener drmSessionEventListener;
```
Comment (joetimmins) [inline core/src/test/java/com/novoda/noplayer/exoplayer/ExoPlayerFacadeTest.java]: static import?

File: core/src/main/java/com/novoda/noplayer/drm/provision/ProvisionExecutorImpl.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package com.novoda.noplayer.drm.provision;
+
+import com.novoda.noplayer.drm.ModularDrmProvisionRequest;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+class ProvisionExecutorImpl implements ProvisionExecutor {
```
Comment (joetimmins) [inline core/src/main/java/com/novoda/noplayer/drm/provision/ProvisionExecutorImpl.java]: Can we call this something that doesn't end in `impl`? `HttpPostingProvisionExecutor` or something maybe?

File: core/src/test/java/com/novoda/noplayer/exoplayer/ExoPlayerFacadeTest.java (hunk 3 above)
Comment (ouchadam) [inline core/src/test/java/com/novoda/noplayer/exoplayer/ExoPlayerFacadeTest.java]: I'm not tooo fussed, it's a test 🍷

**Pull request merged**